pytato/array.py

```diff
@@ -1431,6 +1457,51 @@ def _make_slice(array: Array, starts: Sequence[int], stops: Sequence[int]) -> Ar
     return Slice(array, tuple(starts), tuple(stops))
 
 
+def reshape(array: Array, newshape: Sequence[int]) -> Array:
+    """
+    :param array: array to be reshaped
+    :param newshape: shape of the resulting array
+
+    .. note::
+
+        reshapes of arrays with symbolic shapes not yet implemented.
+    """
+    from pytools import product
+
+    if newshape.count(-1) > 1:
+        raise ValueError("can only specify one unknown dimension")
+
+    if not isinstance(array.size, int):
+        raise ValueError("reshape of arrays with symbolic lengths not allowed")
+
+    newshape_sans_minus_1 = []
```

Comment:
```suggestion
    newshape_explicit = []
```
(Or another name? `sans_minus_1` made me think that the -1 would simply be removed.)

Reply:
d62b4d0